lib/zmq-kernel.coffee

```diff
@@ -0,0 +1,343 @@
+child_process = require 'child_process'
+path = require 'path'
+
+_ = require 'lodash'
+jmp = require 'jmp'
+uuid = require 'uuid'
+zmq = jmp.zmq
+
+Kernel = require './kernel'
+
+
+module.exports =
+class ZMQKernel extends Kernel
+    constructor: (kernelSpec, @grammar, @config, @configPath, onlyConnect = false) ->
+        super kernelSpec
+
+        @executionCallbacks = {}
```

Comment:
Nice catch.

The correct fix is to not set it in the parent. (Execution callbacks are ZMQ-specific, because `jupyter-js-services` does its own callback management via promises)